src/components/WorkoutBlock.tsx

```diff
@@ -32,23 +47,30 @@ const WorkoutBlock: React.FC<WorkoutBlockProps> = ({ editing, workout, updateNam
   }
 
   return (
-    <BlockContainer>
-      <TouchableOpacity
-        accessibilityRole="button"
-        className="flex h-full flex-1 flex-row items-center p-1"
-        onPress={() => onPress(workout)}
-        hitSlop={{ top: 20, bottom: 20, left: 20 }}
-      >
-        <Text className="text-base font-medium">{name}</Text>
-      </TouchableOpacity>
-      <TouchableOpacity
-        accessibilityRole="button"
-        className="flex h-full flex-row items-center justify-center p-1"
-        hitSlop={{ top: 20, bottom: 20, right: 20 }}
-      >
-        <Ionicon name="ellipsis-horizontal" size={16} />
-      </TouchableOpacity>
-    </BlockContainer>
+    <View>
+      <Block
+        title={name}
+        onPress={() => {
+          onPress(workout);
+        }}
+        onOptionsPress={() => {
+          setModalVisible(true);
+        }}
+      />
+      <View>
+        {modalVisible && (
+          <BlockActionsModal
+            deleteWorkout={() => deleteWorkout(workout.id)}
+            setModalVisible={(bool: boolean) => setModalVisible(bool)}
+            renameWorkout={() => setEditing(workout.id)}
+            copyWorkout={() => {
+              editableWorkout.fetch?.();
```

Comment:
Let's avoid fetching the workout here. Also, if we were going to do this, we should use async, await, because otherwise we're creating a race condition by not making sure our fetch finishes before we pass the workout into the `duplicateWorkout` function